src/chronify/sqlalchemy/functions.py

```diff
@@ -18,6 +18,9 @@ def read_database(query: Selectable | str, conn: Connection, schema: TableSchema
             df[config.time_column] = pd.to_datetime(df[config.time_column], utc=True)
         else:
             df[config.time_column] = df[config.time_column].dt.tz_localize("UTC")
+    elif conn.engine.name == "sqlite" and isinstance(config, DatetimeRange):
```

Comment:
refactor so sqlite related processing is in one place. Suggest removing needs_utc_conversion